### ISSUE TITLE
[P1] sizeof/alignof result type hardcoded to ULong — wrong on ILP32 (i686)

### ISSUE THREAD

**Issue by ChaseWNorton** [anthropics/claudes-c-compiler#139]
## Problem

In \`src/frontend/sema/type_checker.rs:178\`, the type checker hardcodes sizeof/alignof as \`CType::ULong\`. On ILP32 targets (i686), sizeof should produce \`CType::UInt\` (4-byte unsigned int), matching size_t.

This causes wrong type propagation in expressions involving sizeof on i686. For example, \`sizeof(x) + 1\` would have its result type computed from ULong instead of UInt, which can affect:
- _Generic selection
- printf format specifier warnings
- Implicit conversion diagnostics
- Type-based optimization decisions

Similarly, \`src/frontend/sema/type_checker.rs:381,391\`: pointer subtraction result type is hardcoded to \`CType::Long\`. On ILP32, ptrdiff_t is \`int\`, not \`long\`.

```c
// On i686:
_Generic(sizeof(int),
    unsigned int: 1,    // GCC selects this
    unsigned long: 2    // CCC selects this (wrong)
)
```

## Files to modify

- \`src/frontend/sema/type_checker.rs\` — Use target-aware size_t/ptrdiff_t types based on pointer size

## Acceptance criteria

- [ ] sizeof/alignof result type is UInt on ILP32, ULong on LP64
- [ ] Pointer subtraction result type is Int on ILP32, Long on LP64
- [ ] \`cargo test --lib\` passes
